rooms.js

```diff
@@ -700,6 +700,29 @@ class GlobalRoom {
 		this.lockdown = true;
 		this.lastReportedCrash = Date.now();
 	}
+	automaticKillRequest() {
+		if (Config.autolockdown === undefined) Config.autolockdown = true; // on by default
+
+		if (Config.autolockdown && Rooms.global.lockdown === true && Rooms.global.battleCount === 0) {
+			// The server is in lockdown, the final battle has finished, and the option is set
+			// so we will now automatically kill the server here if it is not updating.
+			if (Chat.updateServerLock) {
+				for (let roomid of ['development', 'staff', 'upperstaff']) {
+					let curRoom = Rooms(roomid);
+					if (curRoom) curRoom.addRaw('<div class="broadcast-red"><b>Automatic server kill canceled.</b><br /><br />The server tried to automatically kill itself upon the final battle finishing, but the server was updating while trying to kill itself.</div>');
+					curRoom.update();
+				}
+				return;
+			}
+
+			Sockets.workers.forEach(worker => worker.kill());
+
+			// kill server in 10 seconds
+			setTimeout(() => {
+				process.exit();
```

Comment:
What about the workers killing themselves; there's a chance that that could happen without the server restarting if someone turns ``Config.autolockdown`` off in the last 10 seconds...

Should there be a timer here at all? Should it be 10 seconds?  If we check for the Config setting here again, will that over-complicate things?